co-authors-plus.php

```diff
@@ -1048,23 +1048,27 @@ function filter_wp_get_object_terms( $terms, $object_ids, $taxonomies, $args ) {
 	 * @return int Post count
 	 */
 	function filter_count_user_posts( $count, $user_id ) {
-		$user = get_userdata( $user_id );
+		$user     = get_userdata( $user_id );
+		$coauthor = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
 
-		if ( is_object( $user ) ) {
-			$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
+		// Return $count, if no coauthor exists.
+		if ( ! is_object( $coauthor ) ) {
+			return $count;
+		}
 
-			$term = $this->get_author_term( $user );
+		$term = $this->get_author_term( $coauthor );
 
-			if ( $term && ! is_wp_error( $term ) ) {
-				if ( 'guest-author' === $user->type ) {
-					// If using guest author term count, add on linked user count.
-					$count = (int) $count + $term->count;
-				} else {
-					$count = $term->count;
-				}
-			}
+		// Return combined post count, if account is linked.
+		if ( is_object( $term ) && strlen( $coauthor->linked_account ) > 2 ) {
```

Comment:
Getting closer 🙂. I see that we're doing the `is_object()` check twice - once here and once below. Is there a better way to follow the DRY principle?

Reply:
Sure. I could remove one `is_object()` check by using a nested if-else statement. I decided against it for readability reasons. That said, if you prefer DRY over readability, I'm happy to remove the duplicate.

Reply:
Do you see a way to do it without using a nested if-else statement?

Reply:
I could change the order of the checks as follows:

```
if ( ! is_object( $term ) ) {
  return $count;
}

if ( strlen( $coauthor->linked_account ) > 2 ) {
  return $count + $term->count;
}

return $term->count;
```

This way, I would keep the code DRY. The only downside I see, if that `return $term->count;` becomes the fallback value instead of `return $count;`. However, functionality-wise nothing changes.

On the other hand, I could also avoid the else-case by changing the code as follows:

```
if ( is_object( $term ) {
  if ( strlen( $coauthor->linked_account ) > 2 ) ) {
    return $count + $term->count;
  }

  return $term->count;
}
```

While this avoids the else-case, it's still a nested if-statement. While this might be a personal preference, I prefer using [Guard Clauses](https://blog.webdevsimplified.com/2020-01/guard-clauses/) over if-statements.

Reply:
Thanks for sharing different approaches! The guard clause may not be ideal in this case, as we are hooking onto a filter and thus, should return back the original value if no conditions are met. 

I find the second example to be more readable while following the DRY principle.

Reply:
I've just updated the corresponding section according to your feedback. 

The reason why I marked the PR as draft is because I want to add a unit test to this PR, so that we can see immediately if a future change is breaking this PR.